### PR TITLE
Initial ScanAcquisitionManager

### PR DESCRIPTION
**Pull request by agyoungs**

### CODE REVIEW

**Comment by agyoungs** [comment]
Needs testing

File: crs_application/src/task_managers/part_registration_manager.cpp [hunk 1]
```diff
@@ -77,7 +101,19 @@ common::ActionResult PartRegistrationManager::setInput(const datatypes::ScanAcqu
...
+
+  auto localize_result_future = localize_to_part_client_->async_send_request(localize_to_part_request);
+  if (rclcpp::spin_until_future_complete(node_, localize_result_future) !=
+      rclcpp::executor::FutureReturnCode::SUCCESS)
+  {
+      RCLCPP_ERROR(node_->get_logger(), "Localize to Part service call failed");
+      return false;
+  }
+  auto localize_result = localize_result_future.get();
+  // todo(ayoungs) save off transform
```
Comment (jrgnicho) [inline crs_application/src/task_managers/part_registration_manager.cpp:116]: @agyoungs will you be addressing this TODO in this PR?

File: crs_application/src/task_managers/scan_acquisition_manager.cpp [hunk 2]
```diff
@@ -41,7 +41,12 @@ namespace task_managers
 {
 
 ScanAcquisitionManager::ScanAcquisitionManager(std::shared_ptr<rclcpp::Node> node):
-    node_(node)
+    node_(node),
+    scan_positions_(std::vector<geometry_msgs::msg::Transform>()),
+    framos_frame_id_(""),
+    max_time_since_last_point_cloud_(0.1),
```
Comment (jrgnicho) [inline crs_application/src/task_managers/scan_acquisition_manager.cpp:47]: This value should be a global const

File: crs_application/src/task_managers/scan_acquisition_manager.cpp [hunk 3]
```diff
@@ -53,14 +58,33 @@ ScanAcquisitionManager::~ScanAcquisitionManager()
...
+  framos_frame_id_ = node_->declare_parameter("framos_frame_id", "eoat_link");
+  max_time_since_last_point_cloud_ = node_->declare_parameter("max_time_since_last_point_cloud", 0.1);
+
+  // subscribers
+  point_cloud_sub_ = node_->create_subscription<sensor_msgs::msg::PointCloud2>("/point_cloud", 1,
+    std::bind(&ScanAcquisitionManager::handlePointCloud, this, std::placeholders::_1));
+
+  // service client
+  // todo(ayoungs): wait on service?
+  call_freespace_motion_client_ = node_->create_client<crs_msgs::srv::CallFreespaceMotion>("test_plan");
```
Comment (jrgnicho) [inline crs_application/src/task_managers/scan_acquisition_manager.cpp:72]: ideally we would wait on the service to become available and return false if it times out, for now I'd be ok to print a warning at least if it facilitates testing

File: crs_application/src/task_managers/scan_acquisition_manager.cpp [hunk 4]
```diff
@@ -53,14 +58,33 @@ ScanAcquisitionManager::~ScanAcquisitionManager()
...
+
   return true;
 }
 
 common::ActionResult ScanAcquisitionManager::configure(const ScanAcquisitionConfig& config)
 {
-  RCLCPP_WARN(node_->get_logger(),"%s not implemented yet",__PRETTY_FUNCTION__);
-  return true;
+  if (scan_positions_.size() == 0)
```
Comment (jrgnicho) [inline crs_application/src/task_managers/scan_acquisition_manager.cpp:79]: The config should provide the scan positions

File: crs_application/src/task_managers/scan_acquisition_manager.cpp [hunk 5]
```diff
@@ -71,20 +95,69 @@ common::ActionResult ScanAcquisitionManager::verify()
...
+  freespace_motion_request->target_link = framos_frame_id_;
+  freespace_motion_request->goal_pose = scan_positions_.at(scan_index_);
+  freespace_motion_request->execute = true;
+
+  auto result_future = call_freespace_motion_client_->async_send_request(freespace_motion_request);
+  if (rclcpp::spin_until_future_complete(node_, result_future) !=
+    rclcpp::executor::FutureReturnCode::SUCCESS)
+  {
+    RCLCPP_ERROR(node_->get_logger(), "Call Freespace Motion service call failed");
+    // todo(ayoungs): do I need to reset the scan_index_ here and clear the point clouds?
```
Comment (jrgnicho) [inline crs_application/src/task_managers/scan_acquisition_manager.cpp:108]: If motion planning or execution fails we could stop acquisition and store the point clouds that were gathered thus far as long as there's at least one, implementing this logic might require a tweak in the SM

**Review by jrgnicho — COMMENTED** [review]
@agyoungs Looks good for the most part, just have a few question on your TODO's and comments

**Comment by jrgnicho** [comment]
replaced by #19

File: crs_application/src/task_managers/part_registration_manager.cpp (hunk 1 above)
Comment (agyoungs) [reply crs_application/src/task_managers/part_registration_manager.cpp:116]: yes

File: crs_application/src/task_managers/scan_acquisition_manager.cpp (hunk 3 above)
Comment (agyoungs) [reply crs_application/src/task_managers/scan_acquisition_manager.cpp:72]: okay, I'll update accordingly

**Pull request closed without merging**